src/resources/packages/classy/fields/VirtualLocation/ViewingPermissions.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import { Fragment } from 'react';
+import { _x } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import { CheckboxControl, RadioControl } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { METADATA_EVENT_VIRTUAL_SHOW_EMBED_TO } from '../../constants.tsx';
+import useMetaFiltering from './useMetaFiltering.ts';
+
+export default function ViewingPermissions(): JSX.Element {
+	const meta: {
+		showAtRsvpAttendees: boolean;
+		showAtTicketAttendees: boolean;
+	} = useSelect( ( select ) => {
+		const store: {
+			getEditedPostAttribute: ( key: string ) => any;
+		} = select( 'core/editor' );
+		const meta = store.getEditedPostAttribute( 'meta' );
+		const metaValue = meta[ METADATA_EVENT_VIRTUAL_SHOW_EMBED_TO ] ?? [];
+		return {
+			showAtRsvpAttendees: metaValue.includes( 'rsvp' ),
+			showAtTicketAttendees: metaValue.includes( 'ticket' ),
+		};
+	}, [] );
+
+	const [ showAtRsvpAttendees, setShowAtRsvpAttendees ] = useState< boolean >( meta.showAtRsvpAttendees );
+	const [ showAtTicketAttendees, setShowAtTicketAttendees ] = useState< boolean >( meta.showAtTicketAttendees );
+	const [ showOptions, setShowOptions ] = useState< boolean >( showAtRsvpAttendees || showAtTicketAttendees );
+
+	useMetaFiltering(
+		'viewing-permissions',
+		( meta: Object ): Object => {
+			let metaValue = meta[ METADATA_EVENT_VIRTUAL_SHOW_EMBED_TO ] ?? [];
+
+			if ( showAtRsvpAttendees ) {
+				metaValue.push( 'rsvp' );
+			} else {
+				metaValue = metaValue.filter( ( value: string ) => value !== 'rsvp' );
+			}
+
+			if ( showAtTicketAttendees ) {
+				metaValue.push( 'ticket' );
+			} else {
+				metaValue = metaValue.filter( ( value: string ) => value !== 'ticket' );
+			}
+
+			meta[ METADATA_EVENT_VIRTUAL_SHOW_EMBED_TO ] = metaValue;
+
+			return meta;
+		},
+		( meta: Object ): Object => {
```

Comment:
What would you think of allowing for the `onMetaDelete` parameter to be optional, and if not provided it handles this logic automatically?